query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -458,17 +467,22 @@ trait ExecPlan extends QueryCommand {
                   s"execPlan is: $execPlanString, execPlan children ${this.children}")
               srv
           }
-          .map { srv =>
+          .map {
+            case srv: SerializedRangeVector =>
               // fail the query instead of limiting range vectors and returning incomplete/inaccurate results
-              numResultSamples += srv.numRowsSerialized
-              checkSamplesLimit(numResultSamples, querySession.warnings)
-              val srvBytes = srv.estimatedSerializedBytes
-              resultSize += srvBytes
-              querySession.queryStats.getResultBytesCounter(Nil).addAndGet(srvBytes)
-              checkResultBytes(resultSize, querySession.queryConfig, querySession.warnings)
-              srv
+                  numResultSamples += srv.numRowsSerialized
+                  checkSamplesLimit(numResultSamples, querySession.warnings)
+                  val srvBytes = srv.estimatedSerializedBytes
+                  resultSize += srvBytes
+                  querySession.queryStats.getResultBytesCounter(Nil).addAndGet(srvBytes)
+                  checkResultBytes(resultSize, querySession.queryConfig, querySession.warnings)
+                  srv
+            case rv: RangeVector           => rv
```

Comment:
Thinking out loud: does this mean a query like this might not breach any sample-based limits?
```
sum by (_node_) (
    sum_over_time(foo{...}[1000y])
)
```

That would probably be OK if the `dataBytesScanned` limit were tripped, instead, but It looks like that is enforced only in `MultiSchemaPartitionsExec`'s override of `checkResultBytes`-- which Is run only here in this SRV-specific case.

Not 100% sure, but it looks like some obscenely-long-range queries might not trip any limits as long as e.g. series cardinality is low throughout the query range.

Reply:
``dataBytesScanned`` is unaffected. The ``resultBytes`` (bytes serialized) and  ``numResultSamples`` limits won't be breached. IMO, both these are relevant if we move data around, but with target schema, this move is not happening and thus penalizing the query for breaching these limits is not right. The ``dataBytesScanned`` however since unaffected by this change might still be violated.